apollo-federation/src/operation/mod.rs

```diff
@@ -2788,16 +2788,34 @@ impl SelectionSet {
         self.merge_into(std::iter::once(selection_set))
     }
 
-    /// Rebase given `SelectionSet` on self and then inserts it into the inner map. Should any sub
-    /// selection with the same key already exist in the map, the existing selection and the given
-    /// selection are merged, replacing the existing selection while keeping the same insertion index.
+    /// Rebase given `SelectionSet` on self and then inserts it into the inner map. Assumes that given
+    /// selection set does not reference ANY named fragments. If it does, Use `add_selection_set_with_fragments`
+    /// instead.
+    ///
+    /// Should any sub selection with the same key already exist in the map, the existing selection
+    /// and the given selection are merged, replacing the existing selection while keeping the same
+    /// insertion index.
     pub(crate) fn add_selection_set(
         &mut self,
         selection_set: &SelectionSet,
+    ) -> Result<(), FederationError> {
+        self.add_selection_set_with_fragments(selection_set, &NamedFragments::default())
+    }
+
+    /// Rebase given `SelectionSet` on self with the specified fragments and then inserts it into the
+    /// inner map.
+    ///
+    /// Should any sub selection with the same key already exist in the map, the existing selection
+    /// and the given selection are merged, replacing the existing selection while keeping the same
+    /// insertion index.
+    pub(crate) fn add_selection_set_with_fragments(
```

Comment:
I think this can be private, since it's only used by `fn normalize`.

Reply:
I'd keep it as public so it pops out when people try to use other `add_selection_set` that does rebase without fragments. Otherwise it is pretty easy to miss that rebasing wont work.